### PR TITLE
chore(litmus-portal): Project API Integration.

### LINKED ISSUE #1968: Projects Integration and use Apollo GQL client for state management of user data in litmus portal.
Projects Integration, needed bug fixes and state-management optimization for litmus-portal.

### PR DESCRIPTION
**Pull request by ishangupta-ds**
## Proposed changes

Project API Integration and using apollo gql client instead of react-redux for state management of user data.

fixes #1968 

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [x] New feature (non-breaking change which adds functionality)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added necessary documentation (if appropriate)

## Dependency
- None.

## Special notes for your reviewer:
High Priority PR.

### CODE REVIEW

**Comment by ishangupta-ds** [comment]
@gdsoumya @arkajyotiMukherjee @rajdas98 @chandankumar4 @Sanjay1611 @Jonsy13 @Saranya-jena PTAL.

File: litmus-portal/frontend/src/utils/yamlUtils.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ const parsed = (yaml: string) => {
     const parsedYaml = YAML.parse(file as string);
     try {
       const embeddedYaml =
-        parsedYaml.spec.workflowSpec.templates[1].inputs.artifacts[0].raw.data;
+        parsedYaml.spec.templates[1].inputs.artifacts[0].raw.data;
```
Comment (Jonsy13) [inline litmus-portal/frontend/src/utils/yamlUtils.ts:35]: @ishangupta-ds It is changed for parsing new yamls??

Comment (ishangupta-ds) [reply litmus-portal/frontend/src/utils/yamlUtils.ts:35]: We will be using the old yaml template (kind: Workflow) not new one (kind: CronWorkflow) for schedule now option for all the workflows. cc: @ksatchit

File: litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/PersonalDetails/index.tsx [hunk 2]
```diff
@@ -16,10 +18,16 @@ interface personaData {
 // Displays the personals details on the "accounts" tab
 const PersonalDetails: React.FC = () => {
   const classes = useStyles();
-
   const userData: UserData = useSelector((state: RootState) => state.userData);
+  const { username } = userData;
+
+  // Query to get user details
+  const { data } = useQuery<CurrentUserDetails>(GET_USER, {
```
Comment (chandankumar4) [inline litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/PersonalDetails/index.tsx:25]: Can we use react state to get data?

File: litmus-portal/frontend/src/components/Sections/Workflow/WorkflowCluster/index.tsx [hunk 3]
```diff
@@ -62,7 +62,10 @@ const WorkflowCluster: React.FC<WorkflowClusterProps> = ({ gotoStep }) => {
 
   const handleClick = () => {
     getCluster({
-      variables: { project_id: userData.projectID, cluster_type: 'internal' },
+      variables: {
+        project_id: userData.selectedProjectID,
+        cluster_type: 'internal',
```
Comment (chandankumar4) [inline litmus-portal/frontend/src/components/Sections/Workflow/WorkflowCluster/index.tsx:67]: What is meant for internal here?

File: litmus-portal/frontend/src/components/WorkflowCard/CardContent.tsx [hunk 4]
```diff
@@ -87,6 +87,7 @@ const CardContent: React.FC<preDefinedWorkflowData> = ({
           ) : (
             <span />
           )}
+          {description ? description.length < 28 ? <br /> : <div /> : <span />}
```
Comment (chandankumar4) [inline litmus-portal/frontend/src/components/WorkflowCard/CardContent.tsx:90]: What's the use of 28 value here?

File: litmus-portal/frontend/src/containers/layouts/Header/index.tsx [hunk 5]
```diff
@@ -4,20 +4,28 @@ import Toolbar from '@material-ui/core/Toolbar';
...
+import { GET_USER } from '../../../graphql';
 
 const Header: React.FC = () => {
   const classes = useStyles();
-
   const userData: UserData = useSelector((state: RootState) => state.userData);
+  const { username } = userData;
+
+  // Query to get user details
+  const { data } = useQuery<CurrentUserDetails>(GET_USER, {
```
Comment (chandankumar4) [inline litmus-portal/frontend/src/containers/layouts/Header/index.tsx:23]: Can we use react state to fetch data?

File: litmus-portal/frontend/src/pages/HomePage/index.tsx [hunk 6]
```diff
@@ -47,105 +50,116 @@ const CreateWorkflowCard = () => {
 const HomePage = () => {
   const [isOpen, setIsOpen] = useState<boolean>(true);
   const { userData } = useSelector((state: RootState) => state);
-  const { name } = userData;
+  // let name = ''; //data.getUser.name
```
Comment (chandankumar4) [inline litmus-portal/frontend/src/pages/HomePage/index.tsx:53]: if not using then remove it!

File: litmus-portal/frontend/src/pages/HomePage/index.tsx [hunk 7]
```diff
@@ -47,105 +50,116 @@ const CreateWorkflowCard = () => {
...
-      projectID: userData.projectID,
-    },
-    skip: userData.projectID === '',
-    onCompleted: (data) => {
-      user.updateUserDetails({
-        projectName: data.getProject.name,
-      });
-    },
+  // Query to get user details
+  const { data, loading, error } = useQuery<CurrentUserDetails>(GET_USER, {
```
Comment (chandankumar4) [inline litmus-portal/frontend/src/pages/HomePage/index.tsx:59]: Check above comment

File: litmus-portal/frontend/src/pages/HomePage/index.tsx [hunk 8]
```diff
@@ -47,105 +50,116 @@ const CreateWorkflowCard = () => {
...
-                </Typography>
-                <Button className={classes.seeAllBtn}>
-                  <div className={classes.btnSpan}>
-                    <Typography className={classes.btnText}>
-                      {t('home.analytics.moreInfo')}
+    <div>
+      {loading ? (
+        <Loader />
+      ) : error ? (
+        <Typography align="center">Unable to fetch data</Typography>
```
Comment (chandankumar4) [inline litmus-portal/frontend/src/pages/HomePage/index.tsx:96]: Add the text in translation

**Review by chandankumar4 — DISMISSED** [review]

File: litmus-portal/backend/auth/go.sum [hunk 9]
```diff
@@ -13,19 +13,25 @@ github.com/davecgh/go-spew v1.1.1/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSs
 github.com/dgrijalva/jwt-go v3.2.0+incompatible h1:7qlOGliEKZXTDg6OTjfoBKDXWrumCAMpl/TFQ4/5kLM=
 github.com/dgrijalva/jwt-go v3.2.0+incompatible/go.mod h1:E3ru+11k8xSBh+hMPgOLZmtrrCbhqsmaPHjLKYnJCaQ=
 github.com/dgryski/trifles v0.0.0-20190318185328-a8d75aae118c/go.mod h1:if7Fbed8SFyPtHLHbg49SI7NAdJiC5WIA09pe59rfAA=
+github.com/gin-contrib/cors v1.3.1 h1:doAsuITavI4IOcd0Y19U4B+O0dNWihRyX//nn4sEmgA=
```
[the remark this answers is not in the document]
Comment (ishangupta-ds) [reply litmus-portal/backend/auth/go.sum:16]: I just ran the server go.sum automatically changed, without running the auth server its not possible to completely test the projects integration. @chandankumar4

File: litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts [hunk 10]
```diff
@@ -5,10 +5,10 @@ export default [
     urlToIcon:
       'https://hub.litmuschaos.io/api/icon/1.7.0/generic/node-cpu-hog.png',
     chaosWkfCRDLink:
-      'https://raw.githubusercontent.com/litmuschaos/chaos-charts/master/workflows/node-cpu-hog/workflow.yaml',
+      'https://raw.githubusercontent.com/ishangupta-ds/chaos-workflows/master/Argo/argowf-native-node-cpu-hog.yaml',
```
[the remark this answers is not in the document]
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts:8]: No for the time being it will be this, me and @ksatchit are working on raising a PR for the same, will be replacing username with litmuschaos org once the updated workflows are in the org, currently they are in my repo. @chandankumar4

File: litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/PersonalDetails/index.tsx (hunk 2 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/PersonalDetails/index.tsx:25]: No because it was a part of the fix to optimize storage usage on the client, it has been purposely updated to use query from apollo client to avoid replication of data and inefficient memory usage and state management. @chandankumar4

File: litmus-portal/frontend/src/components/Sections/Workflow/WorkflowCluster/index.tsx (hunk 3 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/components/Sections/Workflow/WorkflowCluster/index.tsx:67]: This is required as in the backend we are keeping the cluster type as internal, for connecting external clusters we will be offering this support 1.9 release onwards. (internal denotes the admin cluster or on which GQL server will be running, 'red cluster' as per the convention.) @chandankumar4

File: litmus-portal/frontend/src/components/WorkflowCard/CardContent.tsx (hunk 4 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/components/WorkflowCard/CardContent.tsx:90]: This is required to make the predefined workflow card sizes consistent, it's a necessary fix for the UI. @chandankumar4

File: litmus-portal/frontend/src/containers/layouts/Header/index.tsx (hunk 5 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/containers/layouts/Header/index.tsx:23]: It has been changed to optimize memory and state management. @chandankumar4

File: litmus-portal/frontend/src/pages/HomePage/index.tsx (hunk 6 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/pages/HomePage/index.tsx:53]: Sure.

File: litmus-portal/frontend/src/pages/HomePage/index.tsx (hunk 7 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/pages/HomePage/index.tsx:59]: Its required @chandankumar4

File: litmus-portal/frontend/src/pages/HomePage/index.tsx (hunk 8 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/pages/HomePage/index.tsx:96]: Sure.

**Comment by ishangupta-ds** [comment]
Required changes have been made PTAL @chandankumar4

File: litmus-portal/frontend/src/components/Sections/ChaosWorkflows/BrowseWorkflow/index.tsx [hunk 11]
```diff
@@ -58,21 +61,23 @@ interface SortData {
...
+  const userData: UserData = useSelector((state: RootState) => state.userData);
+  const { selectedProjectID } = userData;
 
   // Query to get workflows
   const { subscribeToMore, data, loading, error } = useQuery<
     Workflow,
     WorkflowDataVars
   >(WORKFLOW_DETAILS, {
-    variables: { projectID: '00002' },
+    variables: { projectID: selectedProjectID as string },
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/Sections/ChaosWorkflows/BrowseWorkflow/index.tsx:72]: is everything is typed correctly we do not need the as string type casting here...WorkflowDataVars already typed projectID as string

File: litmus-portal/frontend/src/components/Sections/ChaosWorkflows/BrowseWorkflow/index.tsx [hunk 12]
```diff
@@ -58,21 +61,23 @@ interface SortData {
...
     fetchPolicy: 'cache-and-network',
   });
 
   // Using subscription to get realtime data
   useEffect(() => {
     subscribeToMore<WorkflowSubscription>({
       document: WORKFLOW_EVENTS,
-      variables: { projectID: '00002' },
+      variables: { projectID: selectedProjectID as string },
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/Sections/ChaosWorkflows/BrowseWorkflow/index.tsx:80]: here too

File: litmus-portal/frontend/src/components/Sections/ChaosWorkflows/ScheduleWorkflow/index.tsx [hunk 13]
```diff
@@ -36,10 +39,13 @@ interface PaginationData {
 const ScheduleWorkflow = () => {
   const classes = useStyles();
 
+  const userData: UserData = useSelector((state: RootState) => state.userData);
+
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/Sections/ChaosWorkflows/ScheduleWorkflow/index.tsx:43]: unnecessary spaces

File: litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/PersonalDetails/index.tsx (hunk 2 above)
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/PersonalDetails/index.tsx:25]: Query variables need to be typed

Comment (arkajyotiMukherjee) [reply litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/PersonalDetails/index.tsx:25]: @chandankumar4 Apollo will serve te data from cache

File: litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/PersonalDetails/index.tsx [hunk 14]
```diff
@@ -16,10 +18,16 @@ interface personaData {
...
+    variables: { username },
+  });
 
-  const { name, email, username } = userData;
+  const name: string = data?.getUser.name as string;
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/PersonalDetails/index.tsx:29]: this typecasting also shouldn't be required...if the interfaces are written correctly

File: litmus-portal/frontend/src/components/WorkflowCard/CardContent.tsx (hunk 4 above)
Comment (arkajyotiMukherjee) [reply litmus-portal/frontend/src/components/WorkflowCard/CardContent.tsx:90]: This is a very bad fix...with absolute numbers in the jsx

File: litmus-portal/frontend/src/containers/app/App.tsx [hunk 15]
```diff
@@ -82,7 +82,7 @@ function App() {
         <div className={classes.root}>
           <div className={classes.appFrame}>
             {/* <Routes /> */}
-            <Routes userData={userData.token} />
+            <Routes userData={userData.token as string} />
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/containers/app/App.tsx:85]: why the unnecessary type casting everywhere?

File: litmus-portal/frontend/src/containers/layouts/Header/index.tsx (hunk 5 above)
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/containers/layouts/Header/index.tsx:23]: Type the query variables

File: litmus-portal/frontend/src/models/user.ts [hunk 16]
```diff
@@ -23,5 +41,7 @@ export type UserAction =
   | UserActionType<typeof UserActions.LOGOUT_USER, string>
   | UserActionType<
       typeof UserActions.UPDATE_USER_DETAILS,
-      { name: string; email: string; projectName: string }
+      {
+        selectedProjectID: string;
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/models/user.ts:45]: The payload needs to be put in an interface if it is not a primitive type like string or number

File: litmus-portal/frontend/src/pages/HomePage/index.tsx [hunk 17]
```diff
@@ -47,105 +50,115 @@ const CreateWorkflowCard = () => {
...
-      projectID: userData.projectID,
-    },
-    skip: userData.projectID === '',
-    onCompleted: (data) => {
-      user.updateUserDetails({
-        projectName: data.getProject.name,
-      });
-    },
+  // Query to get user details
+  const { data, loading, error } = useQuery<CurrentUserDetails>(GET_USER, {
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/pages/HomePage/index.tsx:58]: type query vars here

File: litmus-portal/frontend/src/pages/HomePage/index.tsx [hunk 18]
```diff
@@ -47,105 +50,115 @@ const CreateWorkflowCard = () => {
...
+    variables: { username: userData.username },
   });
 
+  const name: string = data?.getUser.name as string;
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/pages/HomePage/index.tsx:62]: same as above: type casting

**Review by arkajyotiMukherjee — CHANGES_REQUESTED** [review]
Please make the above changes

File: litmus-portal/backend/auth/go.sum (hunk 9 above)
[the remark this answers is not in the document]
Comment (chandankumar4) [reply litmus-portal/backend/auth/go.sum:16]: @Sanjay1611 Please verify is this changes required?

File: litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (chandankumar4) [reply litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts:8]: okay! So for the time being can you push this changes to other branch in chaos-chart

File: litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/PersonalDetails/index.tsx (hunk 2 above)
Comment (chandankumar4) [reply litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/PersonalDetails/index.tsx:25]: okay

File: litmus-portal/frontend/package.json [hunk 19]
```diff
@@ -38,7 +38,7 @@
     "react-plotly.js": "^2.4.0",
     "react-redux": "^7.1.3",
     "react-router-dom": "^5.1.2",
-    "react-scripts": "^3.4.1",
+    "react-scripts": "^3.4.3",
```
Comment (S-ayanide) [inline litmus-portal/frontend/package.json:41]: Changing react-script version from 3.4.1 to 3.4.3 would cause conflicts in package-lock too. Although 3.4.3 is the latest one I think 3.4.1 was good enough to support a lot. Is there any specific reason for upgrading this?

File: litmus-portal/frontend/src/components/WorkflowCard/CardContent.tsx (hunk 4 above)
Comment (S-ayanide) [reply litmus-portal/frontend/src/components/WorkflowCard/CardContent.tsx:90]: If the description is too long it would automatically fit into the provided div with which could be of a customizable width. If you want to shorten up the text to fit the display you can use `string.substring(0, givenLengthToShorten) + "...";` or `lodash` to truncate

**Review by S-ayanide — CHANGES_REQUESTED** [review]

File: litmus-portal/backend/auth/go.sum (hunk 9 above)
[the remark this answers is not in the document]
Comment (Sanjay1611) [reply litmus-portal/backend/auth/go.sum:16]: These changes are do required but wed don't want to have backend changes in this PR. @ishangupta-ds  you can remove it from here. Will include it in a separate PR before RC1

File: litmus-portal/frontend/package.json (hunk 19 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/package.json:41]: I didn't upgrade this, while npm install it automatically updates @S-ayanide do i fix it to 3.4.1 by removing the ^? cc: @arkajyotiMukherjee @gdsoumya @rajdas98 @chandankumar4

**Comment by ishangupta-ds** [comment]
The changes have been made @arkajyotiMukherjee and @S-ayanide , the problem there is the cards are of different sizes due to the length of description, the problem is description is short for some cards and long for some..can you fix this problem after this PR is merged? @S-ayanide it was fine earlier, after refactoring this bug has come up. Though i have added a temporary fix PTAL and fix this issue if you are still working on refactoring / redesigning this component. cc: @gdsoumya @amityt @chandankumar4 @rajdas98

**Review by oumkale — APPROVED** [review]

File: litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/PersonalDetails/index.tsx [hunk 20]
```diff
@@ -22,12 +26,15 @@ const PersonalDetails: React.FC = () => {
   const { username } = userData;
 
   // Query to get user details
-  const { data } = useQuery<CurrentUserDetails>(GET_USER, {
-    variables: { username },
-  });
+  const { data } = useQuery<CurrentUserDetails, CurrentUserDedtailsVars>(
+    GET_USER,
+    {
+      variables: { username },
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/PersonalDetails/index.tsx:32]: remove the comma at the end and make this in one line

File: litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/PersonalDetails/index.tsx [hunk 21]
```diff
@@ -22,12 +26,15 @@ const PersonalDetails: React.FC = () => {
...
+    }
+  );
 
-  const name: string = data?.getUser.name as string;
-  const email: string = data?.getUser.email as string;
+  const name: string = data ? data?.getUser.name : '';
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/PersonalDetails/index.tsx:36]: use javascript nullish operator
```
const name: string = data?.getUser.name ?? '';
```

File: litmus-portal/frontend/src/containers/layouts/Header/index.tsx [hunk 22]
```diff
@@ -20,12 +24,15 @@ const Header: React.FC = () => {
   const { username } = userData;
 
   // Query to get user details
-  const { data } = useQuery<CurrentUserDetails>(GET_USER, {
-    variables: { username },
-  });
+  const { data } = useQuery<CurrentUserDetails, CurrentUserDedtailsVars>(
+    GET_USER,
+    {
+      variables: { username },
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/containers/layouts/Header/index.tsx:30]: remove comma and make in one line

File: litmus-portal/frontend/src/containers/layouts/Header/index.tsx [hunk 23]
```diff
@@ -20,12 +24,15 @@ const Header: React.FC = () => {
...
+    }
+  );
 
-  const name: string = data?.getUser.name as string;
-  const email: string = data?.getUser.email as string;
+  const name: string = data ? data?.getUser.name : '';
+  const email: string = data ? data?.getUser.email : '';
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/containers/layouts/Header/index.tsx:35]: same as above

File: litmus-portal/frontend/src/pages/HomePage/index.tsx [hunk 24]
```diff
@@ -55,11 +55,14 @@ const HomePage = () => {
   const user = useActions(UserActions);
 
   // Query to get user details
-  const { data, loading, error } = useQuery<CurrentUserDetails>(GET_USER, {
+  const { data, loading, error } = useQuery<
+    CurrentUserDetails,
+    CurrentUserDedtailsVars
+  >(GET_USER, {
     variables: { username: userData.username },
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/pages/HomePage/index.tsx:62]: make the other variables like this one...in one line

File: litmus-portal/frontend/src/pages/HomePage/index.tsx [hunk 25]
```diff
@@ -55,11 +55,14 @@ const HomePage = () => {
...
   });
 
-  const name: string = data?.getUser.name as string;
+  const name: string = data ? data?.getUser.name : '';
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/pages/HomePage/index.tsx:65]: same as above

**Review by arkajyotiMukherjee — DISMISSED** [review]
@ishangupta-ds please make these minor corrections

File: litmus-portal/frontend/src/graphql/index.ts [hunk 26]
```diff
@@ -57,16 +57,23 @@ export const GET_USER = gql`
...
+        name
+        id
+      }
+      company_name
+      updated_at
+      created_at
+      removed_at
+      is_email_verified
+      state
+      role
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/graphql/index.ts:76]: Do we need all this information at once? This is almost like a REST call

**Comment by S-ayanide** [comment]
> The changes have been made @arkajyotiMukherjee and @S-ayanide , the problem there is the cards are of different sizes due to the length of description, the problem is description is short for some cards and long for some..can you fix this problem after this PR is merged? @S-ayanide it was fine earlier, after refactoring this bug has come up. Though i have added a temporary fix PTAL and fix this issue if you are still working on refactoring / redesigning this component. cc: @gdsoumya @amityt @chandankumar4 @rajdas98

The description has a width of 60% inside the whole card and the descriptions inside the cards are around 1 to maximum two liners so this seemed pretty okay to me at the moment. However if this is breaking any changes I would suggest increasing the overall height of the card by a few rems and that should do it other than creating a break depending on the length @ishangupta-ds

File: litmus-portal/frontend/src/graphql/index.ts (hunk 26 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/graphql/index.ts:76]: Yeah we need those @S-ayanide

**Comment by ishangupta-ds** [comment]
> > The changes have been made @arkajyotiMukherjee and @S-ayanide , the problem there is the cards are of different sizes due to the length of description, the problem is description is short for some cards and long for some..can you fix this problem after this PR is merged? @S-ayanide it was fine earlier, after refactoring this bug has come up. Though i have added a temporary fix PTAL and fix this issue if you are still working on refactoring / redesigning this component. cc: @gdsoumya @amityt @chandankumar4 @rajdas98
> 
> The description has a width of 60% inside the whole card and the descriptions inside the cards are around 1 to maximum two liners so this seemed pretty okay to me at the moment. However if this is breaking any changes I would suggest increasing the overall height of the card by a few rems and that should do it other than creating a break depending on the length @ishangupta-ds

I have made a temporary fix, who ever is working on the component can debug further. @S-ayanide

File: litmus-portal/backend/auth/go.sum (hunk 9 above)
[the remark this answers is not in the document]
Comment (ishangupta-ds) [reply litmus-portal/backend/auth/go.sum]: I have removed them @Sanjay1611 and @chandankumar4 PTAL.

**Comment by ishangupta-ds** [comment]
changes have been made @arkajyotiMukherjee @S-ayanide @chandankumar4 @Sanjay1611 @rajdas98 @oumkale PTAL.

File: litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts [hunk 27]
```diff
@@ -5,10 +5,10 @@ export default [
...
 
     gitLink:
-      'https://github.com/litmuschaos/chaos-charts/blob/master/workflows/node-cpu-hog/workflow.yaml',
+      'https://github.com/ishangupta-ds/chaos-workflows/blob/master/Argo/argowf-native-node-cpu-hog.yaml',
```
Comment (rajdas98) [inline litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts:11]: Please change the username `ishangupta-ds` to litmuschaos

**Review by rajdas98 — CHANGES_REQUESTED** [review]

File: litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts [hunk 28]
```diff
@@ -20,13 +20,58 @@ export default [
...
+    urlToIcon:
+      'https://hub.litmuschaos.io/api/icon/1.7.0/generic/pod-delete.png',
+    chaosWkfCRDLink:
+      'https://raw.githubusercontent.com/ishangupta-ds/chaos-workflows/master/Argo/argowf-native-pod-delete.yaml',
+
+    gitLink:
+      'https://github.com/ishangupta-ds/chaos-workflows/blob/master/Argo/argowf-native-pod-delete.yaml',
+    provider: 'MayaData',
+    description: 'Deletes a pod',
+    totalRuns: 6700,
```
Comment (rajdas98) [inline litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts:74]: Addition of extra workflow is not a part of this PR. It can go with another PR

**Review by rajdas98 — CHANGES_REQUESTED** [review]

File: litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts (hunk 27 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts:11]: It won't work as these updated charts are not yet in the litmuschaos's chaos-charts repo. Me and @ksatchit are working in my repo and we will be sending a PR soon for adding workflows. These do not exist in litmuschaos org, changing the URL will break the frontend code. @rajdas98 cc: @arkajyotiMukherjee I will update it to litmuschaos when we have the changes in litmuschaos org probably in a couple of days.

File: litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts (hunk 28 above)
Comment (rajdas98) [reply litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts:74]: Let's not change any existing workflow manifest

File: litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts (hunk 27 above)
Comment (rajdas98) [reply litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts:11]: https://github.com/litmuschaos/litmus/pull/1967#discussion_r483437702

File: litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts (hunk 28 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts:74]: Yeah but these are ready so i have added them, just will update the username to litmuschaos in the next PR, and previously we were using the CronWorkflow template, but now we have changed to the Workflow template after discussion with @ksatchit , so we need to make these changes, because the UI has been updated to parse the old template, removing these changes will crash the frontend. @rajdas98

File: litmus-portal/frontend/src/components/WorkflowCard/CardContent.tsx (hunk 4 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/components/WorkflowCard/CardContent.tsx:90]: Yeah for now we have to ahead with this temporary fix, if we wan’t a proper UI. I guess team 3 can make the changes in the design of the cards to fix the UI bug which i fixed temporarily. cc: @chandankumar4

File: litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts (hunk 28 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts]: Removed these changes. @rajdas98

File: litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts (hunk 27 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts]: Have removed these changes. @rajdas98

File: litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts (hunk 28 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts]: With these manifests schedule now is not possible also the workflow run mapping with workflow id. @rajdas98 @gdsoumya cc: @ksatchit

**Comment by ishangupta-ds** [comment]
@arkajyotiMukherjee @gdsoumya @rajdas98 @chandankumar4 @Sanjay1611 the PR has been updated PTAL.

**Review by chandankumar4 — APPROVED** [review]
LGTM Thanks

**Review by Jonsy13 — APPROVED** [review]

**Review by arkajyotiMukherjee — APPROVED** [review]

**Review by oumkale — APPROVED** [review]

**Review by rajdas98 — APPROVED** [review]
